diff --git a/iridiumtk/reassembler/base.py b/iridiumtk/reassembler/base.py
index fa36d5c..656cdc1 100755
--- a/iridiumtk/reassembler/base.py
+++ b/iridiumtk/reassembler/base.py
@@ -3,6 +3,7 @@
 
 import sys
 import datetime
+import time
 import math
 
 from util import base_freq, channel_width, channelize_str, parse_channel
@@ -58,7 +59,7 @@ class MyObject(object):
                 self.level=0
 
         if self.ftype=='p':
-            self.time=float(self.starttime)+self.mstime/1000
+            self.time=time.time()
         elif self.ftype=='j': # deperec
             self.time=self.mstime
             self.timens=int(self.mstime*(10**9))
diff --git a/iridiumtk/reassembler/sbd.py b/iridiumtk/reassembler/sbd.py
index d72f3b7..0c7f76f 100755
--- a/iridiumtk/reassembler/sbd.py
+++ b/iridiumtk/reassembler/sbd.py
@@ -322,7 +322,7 @@ class ReassembleIDASBDACARS(ReassembleIDASBD):
         if q.label == b'_\x7f' and 'nopings' in config.args:
             return
 
-        q.timestamp = datetime.datetime.fromtimestamp(q.time).strftime("%Y-%m-%dT%H:%M:%S%z")
+        q.timestamp = dt.epoch(q.time).isoformat(timespec='seconds')
 
         while len(q.f_reg)>0 and q.f_reg[0:1]==b'.':
             q.f_reg=q.f_reg[1:]
