diff --git a/iridiumtk/reassembler/base.py b/iridiumtk/reassembler/base.py
index fa36d5c..656cdc1 100755
--- a/iridiumtk/reassembler/base.py
+++ b/iridiumtk/reassembler/base.py
@@ -3,6 +3,7 @@
 
 import sys
 import datetime
+import time
 import math
 
 from util import base_freq, channel_width, channelize_str, parse_channel
@@ -58,7 +59,7 @@ class MyObject(object):
                 self.level=0
 
         if self.ftype=='p':
-            self.time=float(self.starttime)+self.mstime/1000
+            self.time=time.time()
         elif self.ftype=='j': # deperec
             self.time=self.mstime
             self.timens=int(self.mstime*(10**9))
